### ISSUE TITLE
Incremental S3 remote backup size and time taken is same as that of full backups

### ISSUE THREAD

**Issue by vishnu-anil** [Altinity/clickhouse-backup#895]
Clickhouse server version: 22.8

```
BACKUPS_TO_KEEP_LOCAL : -1
BACKUPS_TO_KEEP_REMOTE : 60
CLICKHOUSE_HOST : 127.0.0.1
LOG_LEVEL : debug
REMOTE_STORAGE : s3
```

The cronjob runs the --diff-from-remote command and is given below.

```
+ CLICKHOUSE_SERVICES=chi-clickhouse-server-default-0-0
++ date +%Y-%m-%d-%H-%M-%S
+ BACKUP_DATE=2024-04-14-18-00-00
+ declare -A BACKUP_NAMES
+ declare -A DIFF_FROM
+ [[ '' != \a\b\c\d\#\1\2\3\4 ]]
+ CLICKHOUSE_PASSWORD=--password=xxxxxxxxx
+ for SERVER in $CLICKHOUSE_SERVICES
+ [[ 1 == \1 ]]
++ clickhouse-client -q 'SELECT name FROM system.backup_list WHERE location='\''remote'\'' AND name LIKE '\''%full%'\'' AND desc NOT LIKE '\''broken%'\'' ORDER BY created DESC LIMIT 1 FORMAT TabSeparatedRaw' --host=chi-clickhouse-server-default-0-0 --port=9000 --user=clickhouse_user --password=xxxxxxxxxxx
+ LAST_FULL_BACKUP=chi-clickhouse-server-default-0-0-full-2024-04-14-13-45-34
++ clickhouse-client -q 'SELECT name FROM system.backup_list WHERE location='\''remote'\'' AND desc NOT LIKE '\''broken%'\'' ORDER BY created DESC LIMIT 1 FORMAT TabSeparatedRaw' --host=chi-clickhouse-server-default-0-0 --port=9000 --user=clickhouse_user --password=xxxxxxxx
+ PREV_BACKUP_NAME=chi-clickhouse-server-default-0-0-increment-2024-04-14-16-00-00
+ DIFF_FROM[$SERVER]=
++ date +%u
+ [[ 1 == \7 ]]
+ [[ '' == \c\h\i\-\c\l\i\c\k\h\o\u\s\e\-\s\e\r\v\e\r\-\d\e\f\a\u\l\t\-\0\-\0\-\i\n\c\r\e\m\e\n\t\-\2\0\2\4\-\0\4\-\1\4\-\1\6\-\0\0\-\0\0 ]]
+ [[ '' == \c\h\i\-\c\l\i\c\k\h\o\u\s\e\-\s\e\r\v\e\r\-\d\e\f\a\u\l\t\-\0\-\0\-\f\u\l\l\-\2\0\2\4\-\0\4\-\1\4\-\1\3\-\4\5\-\3\4 ]]
+ BACKUP_NAMES[$SERVER]=increment-2024-04-14-18-00-00
+ DIFF_FROM[$SERVER]=--diff-from-remote=chi-clickhouse-server-default-0-0-increment-2024-04-14-16-00-00
+ echo 'set backup name on chi-clickhouse-server-default-0-0 = increment-2024-04-14-18-00-00'
+ for SERVER in $CLICKHOUSE_SERVICES
set backup name on chi-clickhouse-server-default-0-0 = increment-2024-04-14-18-00-00
create increment-2024-04-14-18-00-00 on chi-clickhouse-server-default-0-0
+ echo 'create increment-2024-04-14-18-00-00 on chi-clickhouse-server-default-0-0'
+ clickhouse-client --echo -mn -q 'INSERT INTO system.backup_actions(command) VALUES('\''create chi-clickhouse-server-default-0-0-increment-2024-04-14-18-00-00'\'')' --host=chi-clickhouse-server-default-0-0 --port=9000 --user=clickhouse_user --password=xxxxxxxxx
INSERT INTO system.backup_actions(command) VALUES('create chi-clickhouse-server-default-0-0-increment-2024-04-14-18-00-00')
```

Based on the documentation I understand the backup when given --diff-from-remote takes a full backup but only uploads the diff from the previous backup to the remote. However in my case I see both the incremental and the full backup (taken with the --diff-from-remote) has the same size and ocnsumes same amount of time.

**Comment by Slach** [comment]
increment size depends on now many new data parts do you have
look details in https://github.com/Altinity/clickhouse-backup/blob/master/Examples.md#how-incremental-backups-work-with-remote-storage

Did you use `CronJob` script from
https://github.com/Altinity/clickhouse-backup/blob/master/Examples.md#how-to-use-clickhouse-backup-in-kubernetes?

Looks like shared logs is not the full logs

after INSERT `create` command, `CronJob` shall poll `system.backup_actions` until status "success" or "error"
and run `upload --diff-from-remote...` command

**Comment by vishnu-anil** [comment]
The logs you requested is given below.
```
upload still in progress full-from-incremental-2024-04-15-06-00-00 on chi-clickhouse-server-default-0-0
+ echo 'upload still in progress full-from-incremental-2024-04-15-06-00-00 on chi-clickhouse-server-default-0-0'
+ sleep 5
++ clickhouse-client -mn -q 'SELECT status FROM system.backup_actions WHERE command='\''upload  chi-clickhouse-server-default-0-0-full-from-incremental-2024-04-15-06-00-00'\''' --host=chi-clickhouse-server-default-0-0 --port=9000 --user=clickhouse_user --password=dsfghjkl
+ [[ in progress == success ]]
++ clickhouse-client -mn -q 'SELECT status FROM system.backup_actions WHERE command='\''upload  chi-clickhouse-server-default-0-0-full-from-incremental-2024-04-15-06-00-00'\''' --host=chi-clickhouse-server-default-0-0 --port=9000 --user=clickhouse_user --password=fghjk
+ [[ success != success ]]
+ echo 'BACKUP CREATED'
BACKUP CREATED
```

**Comment by Slach** [comment]
> upload still in progress full-from-incremental-2024-04-15-06-00-00

This is a full backup
I expected logs related to upload command 
`upload chi-clickhouse-server-default-0-0-increment-2024-04-14-18-00-00 --diff-from-remote=chi-clickhouse-server-default-0-0-increment-2024-04-14-16-00-00`

could you share
```
kubectl exec -n <your-namespace> chi-clickhouse-server-default-0-0-0 -c clickhouse-pod -- clickhouse-client -q "SELECT * FROM system.backup_list"
```
?

**Comment by vishnu-anil** [comment]
@Slach thanks for looking into this. Please find the requested output

```
chi-clickhouse-server-default-0-0-full-from-incremental-2024-04-15-10-00-00     2024-04-15 10:00:06     1324153024777   local           regular
chi-clickhouse-server-default-0-0-increment-2024-04-12-10-00-00 1970-01-01 00:00:00     0       remote          broken (can\'t stat metadata.json)
chi-clickhouse-server-default-0-0-increment-2024-04-09-08-00-00 1970-01-01 00:00:00     0       remote          broken (can\'t stat metadata.json)
chi-clickhouse-server-default-0-0-increment-2024-03-01-10-00-00 1970-01-01 00:00:00     0       remote          broken (can\'t stat metadata.json)
chi-clickhouse-server-default-0-0-increment-2023-09-06-05-00-08 1970-01-01 00:00:00     0       remote          broken (can\'t stat metadata.json)
chi-clickhouse-server-default-0-0-increment-2023-06-18-11-00-01 1970-01-01 00:00:00     0       remote          broken (can\'t stat metadata.json)
chi-clickhouse-server-default-0-0-full-from-incremental-2024-03-04-08-00-00     1970-01-01 00:00:00     0       remote          broken (can\'t stat metadata.json)
chi-clickhouse-server-default-0-0-full-2024-04-09-00-00-01      2024-04-09 00:00:09     1286867748030   remote          tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-09-02-00-00 2024-04-09 02:00:09     1287183908898   remote  chi-clickhouse-server-default-0-0-full-2024-04-09-00-00-01      tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-09-04-00-00 2024-04-09 04:00:09     1287478676776   remote  chi-clickhouse-server-default-0-0-increment-2024-04-09-02-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-09-06-00-00 2024-04-09 06:00:08     1287815320159   remote  chi-clickhouse-server-default-0-0-increment-2024-04-09-04-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-09-10-00-00 2024-04-09 10:00:11     1288580028766   remote  chi-clickhouse-server-default-0-0-increment-2024-04-09-06-00-00 tar, regular
chi-clickhouse-server-default-0-0-full-2024-04-10-00-00-00      2024-04-10 00:00:08     1291440409152   remote          tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-10-02-00-00 2024-04-10 02:00:09     1291834309010   remote  chi-clickhouse-server-default-0-0-full-2024-04-10-00-00-00      tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-10-04-00-00 2024-04-10 04:00:09     1292229725357   remote  chi-clickhouse-server-default-0-0-increment-2024-04-10-02-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-10-06-00-00 2024-04-10 06:00:09     1292639200241   remote  chi-clickhouse-server-default-0-0-increment-2024-04-10-04-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-10-08-00-00 2024-04-10 08:00:09     1292955093306   remote  chi-clickhouse-server-default-0-0-increment-2024-04-10-06-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-10-10-00-00 2024-04-10 10:00:09     1293374347153   remote  chi-clickhouse-server-default-0-0-increment-2024-04-10-08-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-10-12-00-00 2024-04-10 12:00:09     1294432552724   remote  chi-clickhouse-server-default-0-0-increment-2024-04-10-10-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-10-14-00-01 2024-04-10 14:00:09     1295811890363   remote  chi-clickhouse-server-default-0-0-increment-2024-04-10-12-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-10-16-00-00 2024-04-10 16:00:09     1297714223360   remote  chi-clickhouse-server-default-0-0-increment-2024-04-10-14-00-01 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-10-18-00-00 2024-04-10 18:00:09     1299943863493   remote  chi-clickhouse-server-default-0-0-increment-2024-04-10-16-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-10-20-00-00 2024-04-10 20:00:13     1300291396147   remote  chi-clickhouse-server-default-0-0-increment-2024-04-10-18-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-10-22-00-00 2024-04-10 22:00:09     1301418156341   remote  chi-clickhouse-server-default-0-0-increment-2024-04-10-20-00-00 tar, regular
chi-clickhouse-server-default-0-0-full-2024-04-11-00-00-00      2024-04-11 00:00:08     1299961663672   remote          tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-11-02-00-00 2024-04-11 02:00:09     1300665368776   remote  chi-clickhouse-server-default-0-0-full-2024-04-11-00-00-00      tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-11-04-00-00 2024-04-11 04:00:09     1301305276376   remote  chi-clickhouse-server-default-0-0-increment-2024-04-11-02-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-11-06-00-00 2024-04-11 06:00:09     1301953987332   remote  chi-clickhouse-server-default-0-0-increment-2024-04-11-04-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-11-08-00-00 2024-04-11 08:00:09     1304155852499   remote  chi-clickhouse-server-default-0-0-increment-2024-04-11-06-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-11-10-00-00 2024-04-11 10:00:09     1304545524425   remote  chi-clickhouse-server-default-0-0-increment-2024-04-11-08-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-11-12-00-00 2024-04-11 12:00:09     1305515545092   remote  chi-clickhouse-server-default-0-0-increment-2024-04-11-10-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-11-14-00-00 2024-04-11 14:00:09     1306299411529   remote  chi-clickhouse-server-default-0-0-increment-2024-04-11-12-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-11-16-00-00 2024-04-11 16:00:14     1307738241462   remote  chi-clickhouse-server-default-0-0-increment-2024-04-11-14-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-11-18-00-00 2024-04-11 18:00:08     1308329852762   remote  chi-clickhouse-server-default-0-0-increment-2024-04-11-16-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-11-20-00-00 2024-04-11 20:00:08     1308758326486   remote  chi-clickhouse-server-default-0-0-increment-2024-04-11-18-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-11-22-00-00 2024-04-11 22:00:08     1306328639105   remote  chi-clickhouse-server-default-0-0-increment-2024-04-11-20-00-00 tar, regular
chi-clickhouse-server-default-0-0-full-2024-04-12-00-00-00      2024-04-12 00:00:07     1307937815340   remote          tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-12-02-00-00 2024-04-12 02:00:07     1308609078687   remote  chi-clickhouse-server-default-0-0-full-2024-04-12-00-00-00      tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-12-04-00-00 2024-04-12 04:00:07     1309008177642   remote  chi-clickhouse-server-default-0-0-increment-2024-04-12-02-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-12-06-00-00 2024-04-12 06:00:08     1309966314518   remote  chi-clickhouse-server-default-0-0-increment-2024-04-12-04-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-12-08-00-00 2024-04-12 08:00:08     1311566122880   remote  chi-clickhouse-server-default-0-0-increment-2024-04-12-06-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-12-12-00-00 2024-04-12 12:00:11     1312803511245   remote  chi-clickhouse-server-default-0-0-increment-2024-04-12-08-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-12-14-00-00 2024-04-12 14:00:06     1313414050328   remote  chi-clickhouse-server-default-0-0-increment-2024-04-12-12-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-12-16-00-00 2024-04-12 16:00:07     1312291394835   remote  chi-clickhouse-server-default-0-0-increment-2024-04-12-14-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-12-18-00-00 2024-04-12 18:00:07     1312824725291   remote  chi-clickhouse-server-default-0-0-increment-2024-04-12-16-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-12-20-00-00 2024-04-12 20:00:07     1313874591421   remote  chi-clickhouse-server-default-0-0-increment-2024-04-12-18-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-12-22-00-00 2024-04-12 22:00:06     1314177141572   remote  chi-clickhouse-server-default-0-0-increment-2024-04-12-20-00-00 tar, regular
chi-clickhouse-server-default-0-0-full-2024-04-13-00-00-00      2024-04-13 00:00:06     1315972043942   remote          tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-13-02-00-00 2024-04-13 02:00:07     1316533032041   remote  chi-clickhouse-server-default-0-0-full-2024-04-13-00-00-00      tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-13-04-00-00 2024-04-13 04:00:06     1317113565581   remote  chi-clickhouse-server-default-0-0-increment-2024-04-13-02-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-13-06-00-01 2024-04-13 06:00:07     1317855863209   remote  chi-clickhouse-server-default-0-0-increment-2024-04-13-04-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-13-08-00-01 2024-04-13 08:00:07     1311625747363   remote  chi-clickhouse-server-default-0-0-increment-2024-04-13-06-00-01 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-13-10-00-01 2024-04-13 10:00:07     1311874690437   remote  chi-clickhouse-server-default-0-0-increment-2024-04-13-08-00-01 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-13-12-00-00 2024-04-13 12:00:06     1312077489728   remote  chi-clickhouse-server-default-0-0-increment-2024-04-13-10-00-01 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-13-14-00-00 2024-04-13 14:00:07     1312981915183   remote  chi-clickhouse-server-default-0-0-increment-2024-04-13-12-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-13-16-00-00 2024-04-13 16:00:07     1314390962807   remote  chi-clickhouse-server-default-0-0-increment-2024-04-13-14-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-13-18-00-00 2024-04-13 18:00:07     1315615606703   remote  chi-clickhouse-server-default-0-0-increment-2024-04-13-16-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-13-20-00-00 2024-04-13 20:00:07     1315576231644   remote  chi-clickhouse-server-default-0-0-increment-2024-04-13-18-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-13-22-00-01 2024-04-13 22:00:06     1316259599948   remote  chi-clickhouse-server-default-0-0-increment-2024-04-13-20-00-00 tar, regular
chi-clickhouse-server-default-0-0-full-2024-04-14-00-00-01      2024-04-14 00:00:06     1317155405101   remote          tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-14-02-00-00 2024-04-14 02:00:07     1317862955777   remote  chi-clickhouse-server-default-0-0-full-2024-04-14-00-00-01      tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-14-04-00-01 2024-04-14 04:00:07     1315283321164   remote  chi-clickhouse-server-default-0-0-increment-2024-04-14-02-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-14-06-00-00 2024-04-14 06:00:07     1316618251640   remote  chi-clickhouse-server-default-0-0-increment-2024-04-14-04-00-01 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-14-08-00-01 2024-04-14 08:00:07     1318247512414   remote  chi-clickhouse-server-default-0-0-increment-2024-04-14-06-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-14-10-00-00 2024-04-14 10:00:07     1318634965031   remote  chi-clickhouse-server-default-0-0-increment-2024-04-14-08-00-01 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-14-12-00-00 2024-04-14 12:00:06     1319308814740   remote  chi-clickhouse-server-default-0-0-increment-2024-04-14-10-00-00 tar, regular
chi-clickhouse-server-default-0-0-full-2024-04-14-13-45-34      2024-04-14 13:45:40     1319896827369   remote          tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-14-16-00-00 2024-04-14 16:00:06     1321258181455   remote  chi-clickhouse-server-default-0-0-full-2024-04-14-13-45-34      tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-14-18-00-00 2024-04-14 18:01:17     1321041519870   remote  chi-clickhouse-server-default-0-0-increment-2024-04-14-16-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-14-20-00-00 2024-04-14 20:00:07     1321938380397   remote  chi-clickhouse-server-default-0-0-increment-2024-04-14-18-00-00 tar, regular
chi-clickhouse-server-default-0-0-increment-2024-04-14-22-00-00 2024-04-14 22:00:07     1322044458869   remote  chi-clickhouse-server-default-0-0-increment-2024-04-14-20-00-00 tar, regular
chi-clickhouse-server-default-0-0-full-2024-04-15-00-00-00      2024-04-15 00:00:06     1320260946778   remote          tar, regular
chi-clickhouse-server-default-0-0-full-from-incremental-2024-04-15-02-00-00     2024-04-15 02:00:06     1321059294255   remote          tar, regular
chi-clickhouse-server-default-0-0-full-from-incremental-2024-04-15-04-00-00     2024-04-15 04:00:07     1321133662165   remote          tar, regular
chi-clickhouse-server-default-0-0-full-from-incremental-2024-04-15-06-00-00     2024-04-15 06:00:06     1322393247733   remote          tar, regular
chi-clickhouse-server-default-0-0-full-from-incremental-2024-04-15-08-00-00     2024-04-15 08:00:06     1323526019201   remote          tar, regular
chi-clickhouse-server-default-0-0-full-from-incremental-2024-04-15-10-00-00     2024-04-15 10:00:06     1324153202815   remote          tar, regular
```

The logs related that you asked for is given below.

```
+ sleep 5
upload still in progress increment-2024-04-16-08-00-01 on chi-clickhouse-server-default-0-0
++ clickhouse-client -mn -q 'SELECT status FROM system.backup_actions WHERE command='\''upload --diff-from-remote=chi-clickhouse-server-default-0-0-increment-2024-04-16-06-00-00 chi-clickhouse-server-default-0-0-increment-2024-04-16-08-00-01'\''' --host=chi-clickhouse-server-default-0-0 --port=9000 --user=clickhouse_user --password=fghjk
upload still in progress increment-2024-04-16-08-00-01 on chi-clickhouse-server-default-0-0
+ [[ in progress == in progress ]]
+ echo 'upload still in progress increment-2024-04-16-08-00-01 on chi-clickhouse-server-default-0-0'
+ sleep 5
++ clickhouse-client -mn -q 'SELECT status FROM system.backup_actions WHERE command='\''upload --diff-from-remote=chi-clickhouse-server-default-0-0-increment-2024-04-16-06-00-00 chi-clickhouse-server-default-0-0-increment-2024-04-16-08-00-01'\''' --host=chi-clickhouse-server-default-0-0 --port=9000 --user=clickhouse_user --password=fghjk
+ [[ in progress == success ]]
++ clickhouse-client -mn -q 'SELECT status FROM system.backup_actions WHERE command='\''upload --diff-from-remote=chi-clickhouse-server-default-0-0-increment-2024-04-16-06-00-00 chi-clickhouse-server-default-0-0-increment-2024-04-16-08-00-01'\''' --host=chi-clickhouse-server-default-0-0 --port=9000 --user=clickhouse_user --password=dfghjk
BACKUP CREATED
+ [[ success != success ]]
+ echo 'BACKUP CREATED'
```

**Comment by Slach** [comment]
Weird, second log shows us backup was successfully created

But `system.backup_list` show nothing backups related to `2024-04-16`

moreover
`chi-clickhouse-server-default-0-0-full-from-incremental`
looks weird name
did you change CronJob scripts?

**Comment by Slach** [comment]
`broken (can\'t stat metadata.json)` means backup was being uploaded but not finished

Could you share `clickhouse-backup print-config` without sensitive credentials?

**Comment by Slach** [comment]
One more thing ;)
could you share logs from clickhouse-backup container?
```
kubectl logs -n <namespace> chi-clickhouse-server-default-0-0-0 -c clickhouse-backup --since=48h
```

**Comment by vishnu-anil** [comment]
Hi @Slach sorry for the delay. Please find the requested details below.

**Recent increment backup log** < Attached
**Clickhouse-backup pod logs**

**backup-configuration**
```
clickhouse@chi-clickhouse-server-default-0-0-0:/$ clickhouse-backup print-config
general:
  remote_storage: s3
  max_file_size: 0
  disable_progress_bar: true
  backups_to_keep_local: -1
  backups_to_keep_remote: 60
  log_level: debug
  allow_empty_backups: true
  download_concurrency: 24
  upload_concurrency: 24
  restore_schema_on_cluster: ""
  upload_by_part: true
  download_by_part: true
  restore_database_mapping: {}
clickhouse:
  username: clickhouse_user
  password: dummmyyyy
  host: 127.0.0.1
  port: 9000
  disk_mapping: {}
  skip_tables:
  - system.*
  - INFORMATION_SCHEMA.*
  - information_schema.*
  - _temporary_and_external_tables.*
  timeout: 5m
  freeze_by_part: false
  freeze_by_part_where: ""
  use_embedded_backup_restore: false
  embedded_backup_disk: ""
  secure: false
  skip_verify: false
  sync_replicated_tables: false
  log_sql_queries: true
  config_dir: /etc/clickhouse-server/
  restart_command: systemctl restart clickhouse-server
  ignore_not_exists_error_during_freeze: true
  check_replicas_before_attach: true
  tls_key: ""
  tls_cert: ""
  tls_ca: ""
  debug: false
s3:
  access_key: ""
  secret_key: ""
  bucket: <hidden>
  endpoint: ""
  region: us-east-2
  acl: private
  assume_role_arn: ""
  force_path_style: true
  path: ""
  disable_ssl: false
  compression_level: 1
  compression_format: tar
  sse: ""
  disable_cert_verification: false
  use_custom_storage_class: false
  storage_class: STANDARD
  concurrency: 1
  part_size: 0
  max_parts_count: 10000
  allow_multipart_download: false
  debug: false
gcs:
  credentials_file: ""
  credentials_json: ""
  bucket: ""
  path: ""
  compression_level: 1
  compression_format: tar
  debug: false
  endpoint: ""
  storage_class: STANDARD
cos:
  url: ""
  timeout: 2m
  secret_id: ""
  secret_key: ""
  path: ""
  compression_format: tar
  compression_level: 1
  debug: false
api:
  listen: 0.0.0.0:7171
  enable_metrics: true
  enable_pprof: false
  username: ""
  password: ""
  secure: false
  certificate_file: ""
  private_key_file: ""
  create_integration_tables: true
  integration_tables_host: ""
  allow_parallel: false
ftp:
  address: ""
  timeout: 2m
  username: ""
  password: ""
  tls: false
  path: ""
  compression_format: tar
  compression_level: 1
  concurrency: 24
  debug: false
sftp:
  address: ""
  port: 22
  username: ""
  password: ""
  key: ""
  path: ""
  compression_format: tar
  compression_level: 1
  concurrency: 1
  debug: false
azblob:
  endpoint_schema: https
  endpoint_suffix: core.windows.net
  account_name: ""
  account_key: ""
  sas: ""
  use_managed_identity: false
  container: ""
  path: ""
  compression_level: 1
  compression_format: tar
  sse_key: ""
  buffer_size: 0
  buffer_count: 3
  max_parts_count: 10000
  timeout: 15m
custom:
  upload_command: ""
  download_command: ""
  list_command: ""
  delete_command: ""
  command_timeout: 4h
  commandtimeoutduration: 4h0m0s
  ```
  
  
[clickhouse-backup.log](https://github.com/Altinity/clickhouse-backup/files/15023849/clickhouse-backup.log)

[incremental-backup.log](https://github.com/Altinity/clickhouse-backup/files/15023890/incremental-backup.log)

**Comment by Slach** [comment]
could you share
`clickhouse-backup version` ?

**Comment by Slach** [comment]
according to shared logs

`default_202309_329357_404155_22_566056.tar`

it means you are applied 566056 mutations to this part

Do you use `ALTER TABLE .. DELETE/ UPDATE .. WHERE ..` or lightweight `DELETE .. WHERE ..`?
If yes, then `clickhouse-server` will produce a lot of new data part and incremental backup will ineffective because calculate increment in data part name level, look details how default incremental backup work
https://github.com/Altinity/clickhouse-backup/blob/master/Examples.md#how-incremental-backups-work-with-remote-storage

I would like to propose upgrading `clickhouse-backup` to the recently released `2.5.0` version and `clickhouse-server` to `24.3`
and use `clickhouse-backup` as a wrapper over `BACKUP/RESTORE` SQL commands with the following config changes:

```yaml
clickhouse:
  use_embedded_backup_restore: true
s3:
  path: "backup"
  object_disk_path: "object_disk_backup"  
``` 

in this case, clickhouse-backup will calculate increment with file checksums level, which will more effective
